client/protocols/autostyle.py

```diff
@@ -17,9 +29,79 @@ def run(self, messages):
         if not messages.get('analytics'):
             log.warning("Autostyle needs to be after analytics")
             return
+        if not messages.get('grading'):
+            log.warning("Autostyle needs to be after grading")
+            return
+        if not self.args.question:
+            log.warning("Autostyle requires a specific question")
+            return
+        messages['autostyle'] = {}
+
+        grading = messages['grading']
+
+        for question in self.args.question:
+            if question in AutoStyleProtocol.ALLOW_QUESTIONS:
+                # Ensure that all tests have passed
+                results = grading.get(question)
+                if not results:
+                    log.warning("No grading info")
+                    return
+                elif results['failed'] or results['locked']:
+                    log.warning("Has not passed all tests")
+                    print("*" * 69)
+                    print(
+                        "To use AutoStyle you must have a correct solution for {0}!".format(question))
+                    print("*" * 69)
+                    return
+            else:
+                log.info("Not an autostyle question")
+                print("*" * 69)
+                print("Make sure to specify -q for a proper AutoStyle-enabled question!")
+                print("*" * 69)
+                return
+
+        print("Once you begin you must finish the experiment in one sitting. This will take at most 2 hours.")
+        confirm = input("Do you wish to continue to AutoStyle? (y/n): ")
+        if confirm.lower().strip() != 'y':
+            return
+
+        messages['analytics']['identifier'] = auth.get_identifier()
+        # Send data to autostyle
+        response_url = self.send_messages(messages, self.SHORT_TIMEOUT)
+        # Parse response_url
+        if response_url:
+            webbrowser.open_new(response_url)
+        else:
+            log.error("There was an error with AutoStyle. Please try again later!")
+
+    def send_messages(self, messages, timeout):
+        """Send messages to server, along with user authentication."""
+        data = {
+            'assignment': self.assignment.endpoint,
+            'messages': messages,
+            'submit': self.args.submit
+        }
+        serialized_data = json.dumps(data).encode(encoding='utf-8')
+        server = 'codestyle.herokuapp.com/ok_launch/'
 
-        messages['analytics']['autostyle'] = 'not ready'
+        address = self.API_ENDPOINT.format(server=server, prefix='http' if self.args.insecure else 'https')
```

Comment:
nit: This is probably better off as it's own variable (108 characters is a bit long)